collector/src/main/java/io/prometheus/jmx/ObjectNameAttributeFilter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2015-2023 The Prometheus jmx_exporter Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.jmx;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+/** Class to implement an ObjectNameAttributeFilter */
```

Comment:
Nitpick: This class has JavaDoc comments that don't add any info that's not already in the class name / method name. I think we should not add comments like this. A comment is great if it has additional info, but if it just repeats the class name or method signature I think it's better to have no comment at all.

Reply:
I agree.

With the addition of Spotless + Google Java Format, Javadoc build warnings occur if public / non-overridden methods don't have proper Javadoc.

I'll improve the Javadoc for these changes; improving other Javadoc as an ongoing interactive process.